### ISSUE TITLE
Dosbox emulated performance vs. real world computers (laptops)

### ISSUE THREAD

**Issue by dodleh** [joncampbell123/dosbox-x#3055]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar feature request(s) already exists?

- [X] I have searched and didn't find any similar issues.

### Is your feature request related to a problem? Please describe.

Dosbox could be improved to better mimic real world systems. This will free up some resources during emulation as Dosbox is too fast in some respects (graphics), while CPU emulation is already costly.

### What you want

To help developers have a better perspective on real world systems and help with testing in similar circumstances.

### Describe alternatives you've considered

Configuring DosBox in number of cycles/second and other parameters offer only a partial solution that does not cover graphics emulation (VGA and SVGA, mostly).

### Additional context

There should be a way of limiting GPU performance with some general guidelines that could work such as a suggested parameter that could limit the number of pixels/second output by the GPU that could cover 2D operations, as 3D operations is even more difficult to imagine, at the moment (there are so many parameters that translate in operations with varying degree of performance in 3D scene setup as well as the separate issue of rendering).

**Comment by dodleh** [comment]
[Dosbox performance vs Real World performance in benchmarks and games 2021.xlsx](https://github.com/joncampbell123/dosbox-x/files/7498573/Dosbox.performance.vs.Real.World.performance.in.benchmarks.and.games.2021.xlsx)

**Comment by Wengier** [comment]
I agree that the CPU cycles are only a partial solution to the problem you stated. DOSBox-X is designed to be highly configurable so I think it is indeed a desired function. You made a nice chart by the way.

**Comment by dodleh** [comment]
The chart hopefully offers a very good reference on how real world systems behave and all tests have been carried by me on my system collection.  Thank you.

The most important result, in my opinion, that shows a clear departure in terms of authenticity is the one on Torque, in Windows 3.x. I have to say that in some DOS games that are also more videocard sensitive (such as QUAKE, if I remember right), the same issue is felt. Even older games such as Lotus, feel a bit smoother than on a real system, which points at a graphical performance that is not well simulated.

Another reason for the utility of the above chart is that it paints a reasonable expectation goal for emulation. As I see it, if emulation resources are handled better, a 300-400 MHz CPU and Voodoo 3 card can be reasonably expected to work on modern systems and around a 166MHz for older systems such as Core 2 Duo CPUs, which would be fantastic.

The only small issue that I have right now in documenting proper performance comparisons is that micro-stuttering and input delays are almost impossible to represent and this is one important long-term goal of emulation.

**Comment by Torinde** [comment]
Would be nice DOSbox-X to become one of the benchmarking tools utilized by sites such as [Anandtech.com](https://www.anandtech.com/) - to populate their [Bench](https://www.anandtech.com/bench/CPU-2020/2758) or just to allow us to populate a table like that above (or [here](https://www.dosbox.com/wiki/Performance)) and to provide answers such as "To emulate Pentium Pro 200MHz you need one core of Ryzen 2GHz". I think if DOSbox-X developers approach some of the CPU benchmarking sites they will gladly join to devise automation, logging, etc. features and [procedures ](https://www.vogons.org/viewtopic.php?t=43280) that would make the process smooth.

Per what I understood:
- CPU emulation is a single-thread algorithm, [unsuitable for splitting](https://github.com/joncampbell123/dosbox-x/issues/1184#issuecomment-514009448), and further speed increases can come either from better host CPU (frequency and IPC) or virtualization instead of emulation
- Video emulation (Glide, 2D as well?) can benefit from multiple threads/cores. Unclear if DOSbox-X Glide/2D emulations currently have multithreaded code?
- remaining activities are not so computation-intensive

Since modern CPUs are multi-core, for simplification we can assume one core is dedicated to CPU emulation with the remaining cores taking care of Video emulation and any other minor tasks. Thus CPU and Video emulation can be benchmarked in isolation, e.g.:
- single thread benchmark: CPU emulation - "To emulate Pentium Pro 200MHz you need one core of Ryzen 2GHz"
- multi-thread benchmark: Video emulation - "To emulate Voodoo 2 you need 3 cores of Ryzen 2.5GHz"
- thus if I want Pentium Pro 200MHz with Voodoo2, then a 4 core Ryzen 2.5GHz is sufficient

Another question, maybe @joncampbell123 - how does the conclusion that CPU emulation is [unsuitable for splitting](https://github.com/joncampbell123/dosbox-x/issues/1184#issuecomment-514009448) squares with the x86 emulation performance by Apple and Microsoft in their OS editions running on ARM? Don't they provide much closer-to-native performance than DOSbox-X? Can some code be leveraged from the [Intel Itanium IA-32_Execution_Layer (LGPL)](https://en.wikipedia.org/wiki/IA-32_Execution_Layer) (LGPL) or the [Itanium 2 emulator](http://ski.sourceforge.net)?
